projects/core/src/cart/facade/active-cart.service.ts

```diff
@@ -33,26 +42,27 @@ import { MultiCartService } from './multi-cart.service';
   providedIn: 'root',
 })
 export class ActiveCartService implements OnDestroy {
-  private readonly PREVIOUS_USER_ID_INITIAL_VALUE =
-    'PREVIOUS_USER_ID_INITIAL_VALUE';
-  private previousUserId = this.PREVIOUS_USER_ID_INITIAL_VALUE;
   private activeCart$: Observable<Cart>;
   protected subscription = new Subscription();
 
-  private userId = OCC_USER_ID_ANONYMOUS;
-  private cartId;
-  private cartUser: User;
-
-  private activeCartId$ = this.store.pipe(
+  // This stream is used for referencing carts in API calls.
+  private activeCartId$ = this.userIdService.getUserId().pipe(
+    // We want to wait with initialization of cartId until we have userId initialized
+    // We have take(1) to not trigger this stream, when userId changes.
+    take(1),
+    switchMapTo(this.store),
     select(MultiCartSelectors.getActiveCartId),
-    filter((cartId) => typeof cartId !== 'undefined'),
+    // We also wait until we initialize cart from localStorage. Before that happens cartId in store === null
+    filter((cartId) => cartId !== null),
     map((cartId) => {
       if (!cartId) {
+        // We fallback to current when we don't have particular cart id -> cartId === '', because that's how you reference latest user cart.
```

Comment:
I'm lost... Shouldn't we then change the condition from `if(!cartId)` to `if(cartId === '')`

Reply:
`null` is filtered above.
In shouldn't be `undefined` here.